permissions/index.html

```diff
@@ -0,0 +1,198 @@
+    <h3>Permissions</h3>
+
+    <div><b>Geolocation:</b> <span id='geolocation'></span></div>
+    <div><b>Notifications:</b> <span id='notifications'></span></div>
+    <div><b>Midi:</b> <span id='midi'></span></div>
+    <div><b>Midi SysEx:</b> <span id='midi-sysex'></span></div>
+    <div><b>Push Notifications:</b> <span id='push-notifications'></span></div>
+
+    <fieldset>
+      <legend>Requests</legend>
+      <button id="getPositionBtn">Geolocation</button>
+      <button id="getNotificationBtn">Notifications</button>
+      <button id="getMidiBtn">Midi SysEx</button>
+      <button id="getPushNotificationBtn">Push Notifications</button>
+    </fieldset>
+
+    <div class="output">
+      <pre id="log"></pre>
+    </div>
+
+    <script>
+      // Service workers require HTTPS (http://goo.gl/lq4gCo). If we're running on a real web server
+      // (as opposed to localhost on a custom port, which is whitelisted), then change the protocol to HTTPS.
+      if ((!location.port || location.port == "80") && location.protocol != 'https:') {
+        location.protocol = 'https:';
+      }
+    </script>
+
+    <script>
+    (function () {
+      function log() {
+        document.querySelector('#log').textContent += Array.prototype.join.call(arguments, '') + '\n';
+      }
+
+      function updatePermission(name, status) {
+        log('update permission for ' + name + ' with ' + status)
+        document.getElementById(name).textContent = status;
+      }
+
+      function requestPushPermission() {
+        navigator.serviceWorker.register('sw-focus.js', {
+          scope: './'
+        }).then(function(reg) {
+          return reg.pushManager.subscribe();
+        }).then(function(reg) {
+          console.log('push reg', reg);
+        });
+      }
+
+      function init() {
+
+        var getPosBtn = document.querySelector('#getPositionBtn');
+        var getNotBtn = document.querySelector('#getNotificationBtn');
+        var getMidBtn = document.querySelector('#getMidiBtn');
+        var getPushNotBtn = document.querySelector('#getPushNotificationBtn');
+
+        // Check for Geolocation API permissions
+        navigator.permissions.query({name:'geolocation'}).then(function(p) {
+          updatePermission('geolocation', p.status);
+
+          p.onchange = function() {
+            updatePermission('geolocation', p.status);
+          };
+        });
+
+        // Check for Notifications API permissions
+        navigator.permissions.query({name:'notifications'}).then(function(p) {
+          updatePermission('notifications', p.status);
+          p.onchange = function() {
+            updatePermission('notifications', p.status);
+          };
+        });
+
+        // Check for Midi/Midi SysEx permissions
+        navigator.permissions.query({name:'midi', sysex:false}).then(function(p) {
+          var update = function() { updatePermission('midi', p.status); };
+          update();
+
+          p.onchange = update;
```

Comment:
In a few of these you use the `p.onchange = update;` formulation, and in a few others you use

    p.onchange = function() {
      updatePermission('notifications', p.status);
    };

Is there a reason that some go with one vs. the other?

In general, what about using

    p.onchange = function() {
      updatePermission('notifications', this.status);
    };

for all of them, switching to using `this` instead of the closure over `p`?